java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java

```diff
@@ -97,7 +97,7 @@ public static Schema convertSchema(org.apache.arrow.flatbuf.Schema schema) {
       String value = kv.value();
       metadata.put(key == null ? "" : key, value == null ? "" : value);
     }
-    return new Schema(Collections2.immutableListCopy(fields), Collections2.immutableMapCopy(metadata));
+    return new Schema(true, Collections.unmodifiableList(fields), Collections.unmodifiableMap(metadata));
```

Comment:
You're skipping the defensive copy here? Seems like a change in behavior.

Reply:
Both defensive copies are done by the constructor itself when the original argument is processed to create a `fields` and a `metadata` collections